### PR TITLE
PMM-12384 Backup artifact metrics

### PR DESCRIPTION
**Pull request by artemgavrilov**
PMM-12384

Link to the Feature Build: SUBMODULES-3370

If this PR adds or removes or alters one or more API endpoints, please review and add or update the relevant [API documents](https://github.com/percona/pmm/tree/main/docs/api) as well:

- [ ] API Docs updated

If this PR is related to some other PRs in this or other repositories, please provide links to those PRs:

- Links to related pull requests (optional).

### CODE REVIEW

File: managed/data/iatemplates/backup_error.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  - name: pmm_backup_error
+    version: 1
+    summary: Backup failed
+    expr: 'pmm_managed_backups_artifacts{status="error"} == bool 1'
+    for: 1m
+    severity: error
+    annotations:
+      description: |-
+        Failed to create backup artifact '{{ $labels.artifact_name}}' on service '{{ $labels.service_id }}'.
+      summary: Failed to create backup artifact '{{ $labels.artifact_name}}' on service '{{ $labels.service_id }}'.
```
Comment (artemgavrilov) [inline managed/data/iatemplates/backup_error.yml:12]: TODO: Find better `summary` and `description`.

File: managed/services/preconditions.go [hunk 2]
```diff
@@ -117,7 +117,7 @@ func CheckArtifactOverlapping(q *reform.Querier, serviceID, locationID, folder s
 	}
 
 	for _, artifact := range artifacts {
-		if artifact.ServiceID != serviceID {
+		if artifact.ServiceID != serviceID { // TODO artifact.ServiceID is empty when service already removed.
```
Comment (artemgavrilov) [inline managed/services/preconditions.go:120]: Will create separate ticket for that

File: managed/services/backup/metrics.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	ctx, cancelCtx := context.WithTimeout(context.Background(), requestTimeout)
+	defer cancelCtx()
+
+	var artifacts []*models.Artifact
+	var services map[string]*models.Service
+	errTx := c.db.InTransactionContext(ctx, nil, func(t *reform.TX) error {
+		var err error
+		artifacts, err = models.FindArtifacts(t.Querier, models.ArtifactFilters{})
+		if err != nil {
+			return errors.Wrapf(err, "failed to find artifacts")
```
Comment (atymchuk) [inline managed/services/backup/metrics.go:73]: nit: It seem like errors.Wrap() is sufficient when no argument

File: managed/services/preconditions.go (hunk 2 above)
Comment (artemgavrilov) [reply managed/services/preconditions.go:120]: https://jira.percona.com/browse/PMM-12448

File: managed/data/iatemplates/backup_error.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+templates:
+  - name: pmm_backup_error
+    version: 1
+    summary: Backup failed
+    expr: 'pmm_managed_backups_artifacts{status="error"} == bool 1'
+    for: 1m
+    severity: error
+    annotations:
+      description: |-
+        Failed to create backup artifact '{{ $labels.artifact_name}}' on service '{{ $labels.service_id }}'.
```
[the remark this answers is not in the document]
Comment (artemgavrilov) [reply managed/data/iatemplates/backup_error.yml]: Done

File: managed/data/iatemplates/backup_error.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (artemgavrilov) [reply managed/data/iatemplates/backup_error.yml]: Done

File: managed/services/backup/metrics.go (hunk 3 above)
Comment (artemgavrilov) [reply managed/services/backup/metrics.go]: Agree, done

File: managed/services/backup/metrics.go [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+		}
+
+		serviceIDs := make([]string, 0, len(artifacts))
+		for _, artifact := range artifacts {
+			serviceIDs = append(serviceIDs, artifact.ServiceID)
+		}
+
+		services, err = models.FindServicesByIDs(t.Querier, serviceIDs)
+		if err != nil {
+			return errors.Wrapf(err, "failed to find services")
```
[the remark this answers is not in the document]
Comment (artemgavrilov) [reply managed/services/backup/metrics.go]: done

File: managed/data/iatemplates/backup_error.yml (hunk 1 above)
Comment (artemgavrilov) [reply managed/data/iatemplates/backup_error.yml]: @rnovikovP @catalinaadam  any suggestions?

**Review by atymchuk — APPROVED** [review]

**Review by idoqo — APPROVED** [review]

**Pull request merged**